src/Client/SignatureVerifier.php

```diff
@@ -53,16 +51,17 @@ public static function createFromRootMetadata(RootMetadata $rootMetadata, bool $
      * @param \Tuf\Metadata\MetadataBase $metadata
      *     The metadata to check signatures on.
      *
-     * @return void
-     *
      * @throws \Tuf\Exception\Attack\SignatureThresholdException
      *   Thrown if the signature threshold has not be reached.
+     * @throws \Tuf\Exception\NotFoundException
+     *   If the metadata role is not recognized.
```

Comment:
The test coverage around this aspect of the code base is, I've found, very sketchy.

We cannot test `checkSignatures()` directly, because it's private (I mean, okay, we could pry it open with reflection if we wanted to, but it's best to avoid that if we can). We test it extensively, though, in UpdaterTest. Indirectly, natch. (I believe that's where we test SignatureTresholdException.) I don't think we have ever had a test case where we're trying to sign with an unknown role, and I can see why not -- it'd be tricky to simulate a scenario like that. Cleaning stuff up might help make that more attainable, though.

I think our test coverage needs to be reorganized in general to keep the technical debt down, but that's a much bigger project.

Reply:
`\Tuf\Client\SignatureVerifier::checkSignatures` is public

Reply:
OH right. It's `checkSingleSignature()` is private. My bad.

Reply:
Should make a follow-up to test the public methods on SignatureVerifier